src/lib/datepicker/calendar.ts

```diff
@@ -29,7 +30,27 @@ import {MatDatepickerIntl} from './datepicker-intl';
 import {MatMonthView} from './month-view';
 import {MatMultiYearView, yearsPerPage} from './multi-year-view';
 import {MatYearView} from './year-view';
+import {ComponentPortal, ComponentType, Portal} from '@angular/cdk/portal';
 
+/** Default header for MatCalendar */
+@Component({
+    moduleId: module.id,
+    selector: 'mat-calendar-header',
+    templateUrl: 'calendar-header.html',
+    encapsulation: ViewEncapsulation.None,
+    preserveWhitespaces: false,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MatCalendarHeader {
+  constructor(@Host() public calendar: MatCalendar<any>,
+              private _intl: MatDatepickerIntl,
+              changeDetectorRef: ChangeDetectorRef) {
+
+    this._intlChanges = _intl.changes.subscribe(() => changeDetectorRef.markForCheck());
```

Comment:
You need to make sure this stream gets cleaned up when the component is destroyed. The standard way we do this in other components is to create a stream that emits on destroy and then use the `takeUntil` operator when we're subscribing. Here's an example:

Create a `_destroyed` stream:
https://github.com/angular/material2/blob/e7533a5dc296ed9bd570a7af477fddd3da7da255/src/cdk/tree/padding.ts#L25
Emit on the `_destroyed` stream when component is destroyed: https://github.com/angular/material2/blob/e7533a5dc296ed9bd570a7af477fddd3da7da255/src/cdk/tree/padding.ts#L57-L60
Use `takeUntil(this._destroyed)` when subscribing: https://github.com/angular/material2/blob/e7533a5dc296ed9bd570a7af477fddd3da7da255/src/cdk/tree/padding.ts#L53

Reply:
Thanks for the hint, I will try to fix it.

Reply:
added in a88fecbec2c3292017613bf4e75c38ff7c632042

hope this fixes it

Reply:
@mmalerba Or does it need to be? 

```
/** Default header for MatCalendar */
@Component({
    moduleId: module.id,
    selector: 'mat-calendar-header',
    templateUrl: 'calendar-header.html',
    encapsulation: ViewEncapsulation.None,
    preserveWhitespaces: false,
    changeDetection: ChangeDetectionStrategy.OnPush,
})
export class MatCalendarHeader implements OnDestroy {
  /** Subject that emits when the component has been destroyed. */
  private _destroyed = new Subject<void>();

  constructor(@Host() public calendar: MatCalendar<any>,
              private _intl: MatDatepickerIntl,
              changeDetectorRef: ChangeDetectorRef,
              @Optional() private _dir: Directionality) {
    if (this._dir) {
      this._dir.change.pipe(takeUntil(this._destroyed)).subscribe(
          () => changeDetectorRef.markForCheck()
      );
    }
  }

  ngOnDestroy() {
    this._destroyed.next();
    this._destroyed.complete();
  }

}
```

Reply:
Or do I just have to add this 

```
ngOnDestroy() {
    this._destroyed.next();
    this._destroyed.complete();
    this._intlChanges.unsubscribe();
  }
```